Userland/Libraries/LibJIT/GDB.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023, Jesús Lapastora <cyber.gsuscode@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Checked.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <LibJIT/GDB.h>
+
+namespace JIT::GDB {
+
+// Declarations from https://sourceware.org/gdb/current/onlinedocs/gdb.html/Declarations.html#Declarations.
+enum class JitActions : u32 {
+    NoAction = 0,
+    RegisterFn = 1,
+    UnregisterFn = 2,
+};
+
+struct JitCodeEntry {
+    RawPtr<JitCodeEntry> next_entry;
+    RawPtr<JitCodeEntry> prev_entry;
+    RawPtr<char const> symfile_addr;
+    u64 symfile_size;
+};
+
+struct JitDescriptor {
+    u32 version;
+    JitActions action_flag;
+    RawPtr<JitCodeEntry> relevant_entry;
+    RawPtr<JitCodeEntry> first_entry;
```

Comment:
```suggestion
    JitCodeEntry* next_entry;
    JitCodeEntry* prev_entry;
    char const* symfile_addr;
    u64 symfile_size;
};

struct JitDescriptor {
    u32 version;
    JitActions action_flag;
    JitCodeEntry* relevant_entry;
    JitCodeEntry* first_entry;
```

I'm pretty sure RawPtr is only meant for usage with certain templates